### PR TITLE
Remove JSON Compact Serialisation

### PR DESCRIPTION
**Pull request by kylef**
I wan't to open this discussion. I don't think JSON Compact adds much value and I don't think it's worth keeping around:

- It's pretty complicated format and it's not as readable and clear as JSON Serialisation.
- It doesn't add that much space saving for JSON Serialisation and GZIP would ultimately solve a lot of the size problems of the JSON Serialisation.

Would love to hear the opinions of others here.

### CODE REVIEW

**Comment by smizell** [comment]
I agree. It is one more thing to maintain that no one is using. It can always be added back if necessary.

**Comment by w-vi** [comment]
I don't agree that it is complicated I would actually argue that it is much simpler then JSON but I agree that it is useless at the moment and we should get rid of it.

**Comment by smizell** [comment]
@w-vi the compact format speaks to the soul of Lisp fans :)

**Comment by kylef** [comment]
> I don't agree that it is complicated

Looking at a large element structure consisting of dozens of `[[[[[[[[[[[[[` brackets it's hard to follow and read and follow.

**Comment by w-vi** [comment]
@kylef is it really for human consumption though?

**Comment by kylef** [comment]
I'd say so:

```json
[
  "parseResult",
  null,
  null,
  [
    [
      "category",
      {
        "title": [
          "string",
          null,
          null,
          "My API"
        ],
        "classes": [
          "array",
          null,
          null,
          [
            [
              "string",
              null,
              null,
              "api"
            ]
          ]
        ]
      },
      null,
      [
        [
          "copy",
          null,
          null,
          "Hello World"
        ],
        [
          "category",
          {
            "classes": [
              "array",
              null,
              null,
              [
                [
                  "string",
                  null,
                  null,
                  "resourceGroup"
                ]
              ]
            ]
          },
          null,
          [
            [
              "resource",
              {
                "title": [
                  "string",
                  null,
                  null,
                  "User"
                ]
              },
              {
                "href": [
                  "string",
                  null,
                  null,
...
```

**Comment by kylef** [comment]
@w-vi I believe that was one of the intentions, is that it is easier to write by hand.

**Comment by w-vi** [comment]
If the intention was that compact is human friendly then it failed :)

**Review by pksunkara — APPROVED** [review]
